### PR TITLE
Make TensorFlow demo compatible with all the current EESSI repos and versions

### PR DESCRIPTION
**Pull request by laraPPr**

### CODE REVIEW

**Comment by boegel** [comment]
Maybe somewhat shorter & easier to digest:

```shell
if [[ ${EESSI_CVMFS_REPO} == "/cvmfs/software.eessi.io" ]] && [[ ${EESSI_VERSION} == "2023.06" ]]; then module load TensorFlow/2.13.0-foss-2023a
elif [[ ${EESSI_CVMFS_REPO} == "/cvmfs/pilot.eessi-hpc.org" ]] && [[ ${EESSI_PILOT_VERSION} == "2021.12" ]]; then module load TensorFlow/2.3.1-foss-2020a-Python-3.8.2
else echo "Don't know which TensorFlow module to load for ${EESSI_CVMFS_REPO}/versions/${EESSI_VERSION}${EESSI_PILOT_VERSION}" >&2; exit 1; fi
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: TensorFlow/run.sh [hunk 1]
```diff
@@ -1,5 +1,21 @@
 #!/bin/bash
+set -e
+
+echo $EESSI_CVMFS_REPO
+if [ $EESSI_CVMFS_REPO == "/cvmfs/software.eessi.io" ]; then
+    echo $EESSI_VERSION
+    if [ $EESSI_VERSION == "2023.06" ]; then
+        module load TensorFlow/2.13.0-foss-2023a
+    fi
```
[the remark this answers is not in the document]
Comment (laraPPr) [reply TensorFlow/run.sh]: Changed it to the suggested change that you mention [here](https://github.com/EESSI/eessi-demo/pull/31#issuecomment-1932558286). So than unknown EESSI version should be covered. Or do you want something else?

[the remark this answers is not in the document]
Comment (boegel) [reply TensorFlow/run.sh]: Indeed, the `else` takes care of unknown versions now

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Tested, works like a charm! :+1:

**Pull request merged**